internal/wasm/engine.go

```diff
@@ -1,20 +1,26 @@
 package internalwasm
 
+import "io"
+
 // Engine is a Store-scoped mechanism to compile functions declared or imported by a module.
 // This is a top-level type implemented by an interpreter or JIT compiler.
 type Engine interface {
 	// NewModuleEngine compiles down the function instances in a module, and returns ModuleEngine for the module.
 	//
+	// * name is the name the module was instantiated with used for error handling.
 	// * importedFunctions: functions this module imports, already compiled in this engine.
 	// * moduleFunctions: functions declared in this module that must be compiled.
 	//
 	// Note: Input parameters must be pre-validated with internalwasm.Module Validate, to ensure no fields are invalid
 	// due to reasons such as out-of-bounds.
-	NewModuleEngine(importedFunctions, moduleFunctions []*FunctionInstance) (ModuleEngine, error)
+	NewModuleEngine(name string, importedFunctions, moduleFunctions []*FunctionInstance) (ModuleEngine, error)
 }
 
 // ModuleEngine implements function calls for a given module.
 type ModuleEngine interface {
+	// Closer releases the resources allocated by functions in this ModuleEngine.
+	io.Closer
```

Comment:
ps this gives us space to use errors later as needed even if we currently implement close async via finalizer.

Reply:
just curious: is it a common practice in Golang for a non-IO interface to adopt `io.Closer` because it has `Close() error`?